### PR TITLE
100% test coverage blog post

### PR DESCRIPTION
**Pull request by oluoluoxenfree**
## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

## Show me
_Provide screenshots/animated gifs/videos if necessary._

# REMEMBER: Attach this PR to the Trello card

### CODE REVIEW

File: content/blog/2022/100-percent-test-coverage.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+These dimensions mean that 100% unit test coverage is rarely actually covering
+every aspect that it could in a codebase. If there was a visual regression in
+one of our codebases with 100% test coverage, for example, our tests wouldn’t
+record this. Visual regression tests don’t have a coverage metric, and can be
+unwieldy when used on large codebases due to the number of screenshots that have
+to be reviewed, but cover a part of the codebase not easily tested by other
+means.
+
+Recently I’ve been working on polyfills for bleeding edge CSS features, and they
```
Comment (jgerigmeyer) [inline content/blog/2022/100-percent-test-coverage.md:41]: Could we link to one or some of the "polyfills for bleeding edge CSS features" you've been working on?

File: content/blog/2022/100-percent-test-coverage.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+need a very different method of testing compared to what I am used to. [Web
+Platform Tests], which test that browsers are all implementing features in the
+same way, do not have a coverage metric and are integration tests for the most
+part. CSS is the main thing compared using JS, and there are visual regression
+and manual tests in the mix. Using PostCSS also exposed me to their test runner,
+tape, which compares CSS files with an expectation file, and also does not have
```
Comment (jgerigmeyer) [inline content/blog/2022/100-percent-test-coverage.md:47]: Could link to tape?

**Review by jgerigmeyer — COMMENTED** [review]
Looks good! I think it could use a banner image/photo, and ideally even a few images throughout the article (screenshots of various coverage outputs?)

@SondraE Do you have ideas?

**Comment by SondraE** [comment]
> Looks good! I think it could use a banner image/photo, and ideally even a few images throughout the article (screenshots of various coverage outputs?)
> 
> @SondraE Do you have ideas?

@oluoluoxenfree @jgerigmeyer Yes, I agree that we should include a header image (1600px w), if at all possible. It could be a humorous one like Mia used in this post (https://www.oddbird.net/2021/05/14/support-unknown/) or maybe a code sample if there's a way to make it visually interesting (https://www.oddbird.net/2017/09/25/susy-use/) or something a simple as this free image (https://pixabay.com/illustrations/hundred-one-numbers-100-percent-751063/).

Code samples throughout would be fantastic. 

Is there any way to show the test suite catching an error? (I don't know enough about it to know whether that's possible to show in a screenshot.)

**Review by SondraE — CHANGES_REQUESTED** [review]
> Looks good! I think it could use a banner image/photo, and ideally even a few images throughout the article (screenshots of various coverage outputs?)
> 
> @SondraE Do you have ideas?

@oluoluoxenfree @jgerigmeyer Yes, I agree that we should include a header image (1600px w), if at all possible. It could be a humorous one like Mia used in this post (https://www.oddbird.net/2021/05/14/support-unknown/) or maybe a code sample if there's a way to make it visually interesting (https://www.oddbird.net/2017/09/25/susy-use/) or something a simple as this free image (https://pixabay.com/illustrations/hundred-one-numbers-100-percent-751063/).

Code samples throughout would be fantastic. Is there any way to show the test suite catching an error? (I don't know enough about it to know whether that's possible to show in a screenshot.)

**Comment by oluoluoxenfree** [comment]
[100% Test Coverage Blog Post](https://trello.com/c/gfBIAx8b/71-100-test-coverage-blog-post)

File: content/blog/2022/100-percent-test-coverage.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
+---
+title: What's the impact of aiming for 100% test coverage?
+author: olu
+date: 2022-05-24
```
Comment (jgerigmeyer) [inline content/blog/2022/100-percent-test-coverage.md:4]: We should remember to update this right before merging.

**Review by jgerigmeyer — APPROVED** [review]

**Review by SondraE — APPROVED** [review]
lgtm!

**Review by jerivas — APPROVED** [review]
Lookin' good 👍

File: content/blog/2022/100-percent-test-coverage.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+tests. It usually refers to unit tests or integration tests. There are automated
+ways of measuring it, and these usually present your coverage as a percentage.
+100% test coverage is a contentious issue. At OddBird we often aim for this
+level of test coverage, and after working on a few different projects I thought
+it would be good to assess how I find it and the pros and cons of testing in
+this way.
+
+{{ embed.img(
+  src='blog/2022/metecho-coverage.png',
+  alt='One hundred percent test coverage on the Metecho project'
```
Comment (sanajaved7) [inline content/blog/2022/100-percent-test-coverage.md:29]: Maybe we could add some additional description to the alt text of what the image shows? Something like: `Image of the Metecho project's test coverage summary which shows 100% test coverage for statements, branches, functions and lines. Also displayed is the number of test suites, tests, snapshots and the time it took to run the tests`

File: content/blog/2022/100-percent-test-coverage.md [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+tests are left to the end. It’s often easier to use a test driven approach
+without a code coverage target; writing tests that are embedded in the code
+writing process means they cannot be skipped, and not having a target means that
+there is less infilling to do at the end of writing code.
+
+## Conclusion
+
+Test coverage is one tool in our toolbox, and having a metric to measure it
+gives some reassurance that we have looked at the code from many angles. The
+other testing dimensions are covered in our testing processes -- we often test
```
Comment (sanajaved7) [inline content/blog/2022/100-percent-test-coverage.md:145]: The conclusion looks great! 🎉  One suggestion here is to break up this sentence that starts with `The other testing dimensions...` - it was a bit hard to follow since it was quite long.

**Review by sanajaved7 — APPROVED** [review]
Requested two minor changes but this looks great! 💯

**Review by jgerigmeyer — APPROVED** [review]
@oluoluoxenfree I think you could update the publish date to today/tomorrow and merge accordingly?

**Comment by oluoluoxenfree** [comment]
@jgerigmeyer i thought i had set it to today, checking! if so i'll publish it now :)
